### DISCUSSION TITLE
Pluggable observability?

### DISCUSSION THREAD

**Discussion by jdunck** [hynek/stamina#27] [Ideas]
At current work, we use datadog, not prometheus.   At previous work, we used an internal metrics tool.

Since the `_instrumentation` needs are pretty modest, how would you feel about defining an "observation sink" protocol and having a way for `stamina` to be configured with a dynamic callable or module to receive the observable event?

I see the implementation already uses an `INSTRUMENTS` list, so I think this might just mean stabilizing the interface and documenting how to add to `INSTRUMENTS`?

**Comment by jdunck** [comment]
If we can agree on a design target I'm happy to implement.

**Comment by hynek** [comment]
I was wondering how long it'll take for someone to ask. :)

I just wanted to keep the next "big" release small. It would be useful if you could provide me with metadata you definitely want to receive, in case I might miss it. There is already the definition of `RetryHook` in https://github.com/hynek/stamina/blob/main/src/stamina/typing.py but at this point we still can change it.

**Comment by hynek** [comment]
Landed in #37! 🥳
